references/plugins/capcenter.go

```diff
@@ -68,66 +86,100 @@ func NewCenterClient(ctx context.Context, name, address, token string) (CenterCl
-	url, err := url.Parse(addr)
+func Parse(addr string) (string, *Content, error) {
+	URL, err := url.Parse(addr)
 	if err != nil {
 		return "", nil, err
 	}
-	l := strings.Split(strings.TrimPrefix(url.Path, "/"), "/")
-	switch url.Host {
-	case "github.com":
-		// We support two valid format:
-		// 1. https://github.com/<owner>/<repo>/tree/<branch>/<path-to-dir>
-		// 2. https://github.com/<owner>/<repo>/<path-to-dir>
-		if len(l) < 3 {
-			return "", nil, errors.New("invalid format " + addr)
-		}
-		if l[2] == "tree" {
-			// https://github.com/<owner>/<repo>/tree/<branch>/<path-to-dir>
-			if len(l) < 5 {
+	l := strings.Split(strings.TrimPrefix(URL.Path, "/"), "/")
+	switch URL.Scheme {
+	case "http", "https":
+		switch URL.Host {
+		case "github.com":
+			// We support two valid format:
+			// 1. https://github.com/<owner>/<repo>/tree/<branch>/<path-to-dir>
+			// 2. https://github.com/<owner>/<repo>/<path-to-dir>
+			if len(l) < 3 {
 				return "", nil, errors.New("invalid format " + addr)
 			}
-			return TypeGithub, &GithubContent{
-				Owner: l[0],
-				Repo:  l[1],
-				Path:  strings.Join(l[4:], "/"),
-				Ref:   l[3],
-			}, nil
+			if l[2] == "tree" {
+				// https://github.com/<owner>/<repo>/tree/<branch>/<path-to-dir>
+				if len(l) < 5 {
+					return "", nil, errors.New("invalid format " + addr)
+				}
+				return TypeGithub, &Content{
+					GithubContent: GithubContent{
+						Owner: l[0],
+						Repo:  l[1],
+						Path:  strings.Join(l[4:], "/"),
+						Ref:   l[3],
+					},
+				}, nil
+			}
+			// https://github.com/<owner>/<repo>/<path-to-dir>
+			return TypeGithub, &Content{
+					GithubContent: GithubContent{
+						Owner: l[0],
+						Repo:  l[1],
+						Path:  strings.Join(l[2:], "/"),
+						Ref:   "", // use default branch
+					},
+				},
+				nil
+		case "api.github.com":
+			if len(l) != 5 {
+				return "", nil, errors.New("invalid format " + addr)
+			}
+			//https://api.github.com/repos/<owner>/<repo>/contents/<path-to-dir>
+			return TypeGithub, &Content{
+					GithubContent: GithubContent{
+						Owner: l[1],
+						Repo:  l[2],
+						Path:  l[4],
+						Ref:   URL.Query().Get("ref"),
+					},
+				},
+				nil
+		default:
 		}
-		// https://github.com/<owner>/<repo>/<path-to-dir>
-		return TypeGithub, &GithubContent{
-			Owner: l[0],
-			Repo:  l[1],
-			Path:  strings.Join(l[2:], "/"),
-			Ref:   "", // use default branch
+	case "oss":
+		bucket := strings.Split(URL.Host, ".")[0]
+		entryPoint := strings.TrimPrefix(URL.Host, bucket+".")
+
+		return TypeOss, &Content{
+			OssContent: OssContent{
+				EntryPoint: entryPoint,
+				Bucket:     bucket,
+			},
 		}, nil
-	case "api.github.com":
-		if len(l) != 5 {
-			return "", nil, errors.New("invalid format " + addr)
-		}
-		//https://api.github.com/repos/<owner>/<repo>/contents/<path-to-dir>
-		return TypeGithub, &GithubContent{
-			Owner: l[1],
-			Repo:  l[2],
-			Path:  l[4],
-			Ref:   url.Query().Get("ref"),
+	case "file":
```

Comment:
`local` be better?

Reply:
I can't say. I use `file://` because when enter that in your browser like `file:///Users/`, you'll really see it's there. It's like a   custom?